core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/score/director/InnerScoreDirector.java

```diff
@@ -292,12 +292,48 @@
 
     void afterElementRemoved(ListVariableDescriptor<Solution_> variableDescriptor, Object entity, int index);
 
-    void beforeElementMoved(ListVariableDescriptor<Solution_> variableDescriptor,
-            Object sourceEntity, int sourceIndex,
-            Object destinationEntity, int destinationIndex);
+    /**
+     * Notify the score director before a subList of a list variable changes. The subList is a continuous sequence of
+     * the list variable elements starting at {@code fromIndex} (inclusive) and ending at {@code toIndex} (exclusive).
+     * <p>
+     * The subList has to comply with the following contract:
+     * <ol>
+     * <li>{@code fromIndex} must be greater than or equal to 0; {@code toIndex} must be less than or equal to the list variable
+     * size.</li>
+     * <li>{@code toIndex} must be greater than or equal to {@code fromIndex}.</li>
+     * <li>The subList must contain all elements that are going to be changed by the move.</li>
+     * <li>The subList is allowed to contain elements that are not going to be changed by the move.</li>
+     * <li>The subList may be empty ({@code fromIndex} equals {@code toIndex}) if none of the existing list variable elements
+     * are going to be changed by the move.</li>
+     * </ol>
+     *
+     * @param variableDescriptor descriptor of the variable being changed
+     * @param entity the entity owning the variable being changed
+     * @param fromIndex low endpoint (inclusive) of the subList
+     * @param toIndex high endpoint (exclusive) of the subList
+     */
+    void beforeSubListChanged(ListVariableDescriptor<Solution_> variableDescriptor, Object entity, int fromIndex, int toIndex);
```

Comment:
Naming must align to ListVariableListener interface, so maybe before/afterListVariableChanged(...)